src/main/java/org/opentripplanner/middleware/controllers/api/ApiKeyManagementController.java

```diff
@@ -0,0 +1,150 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import com.amazonaws.services.apigateway.model.CreateApiKeyResult;
+import com.amazonaws.services.apigateway.model.GetUsageResult;
+import org.eclipse.jetty.http.HttpStatus;
+import org.opentripplanner.middleware.models.ApiUser;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.persistence.TypedPersistence;
+import org.opentripplanner.middleware.utils.ApiGatewayUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.List;
+
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsText;
+import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
+
+/**
+ * Sets up HTTP endpoints for managing api keys and viewing api key request logs.
+ */
+public class ApiKeyManagementController {
+
+    private static TypedPersistence<ApiUser> apiUsers = Persistence.apiUsers;
+
+    private static final String DEFAULT_USAGE_PLAN_ID
+        = getConfigPropertyAsText("DEFAULT_USAGE_PLAN_ID");
+
+    /**
+     * Register http endpoints with {@link spark.Spark} instance at the provided API prefix.
+     */
+    public static void register(Service spark, String apiPrefix) {
+        // FIXME need to agree on endpoint (/admin or /secure or a mix) and endpoint names
+        spark.get(apiPrefix + "/admin/apikey/create", ApiKeyManagementController::createApiKeyForApiUser, JsonUtils::toJson);
+        spark.get(apiPrefix + "/admin/apikey/delete", ApiKeyManagementController::deleteApiKeyForApiUser, JsonUtils::toJson);
+        spark.get(apiPrefix + "/admin/apikey/usage", ApiKeyManagementController::getUsageLogsForApiUser, JsonUtils::toJson);
```

Comment:
@br648, I think these should ultimately go under `/secure`. For ApiUsers creating keys directly, we'll probably only let them create up to some limit (e.g., 2 keys), but that might not apply for admins. Same goes for usage plan. ApiUsers can only create for the DEFAULT_USAGE_PLAN, but an admin user could pass in a query param for a different plan (which would presumably have higher request limits).

Reply:
Updated.